BridgeAppSDK/SBAGenericStepViewController.swift

```diff
@@ -0,0 +1,951 @@
+    }
+    
+    
+    // Override result so we can generate our own result from our internal model
+    override open var result: ORKStepResult? {
+        get { return customResult() }
+    }
+    
+    /**
+     Static method to determine if this view controller class supports the provided step. This will vary
+     based on the 'ORKAnswerFormat' and 'ORKQuestionType' for each of the 'ORKFormItems' in the step.
+     */
+    static open func doesSupport(_ step: ORKStep) -> Bool {
+                
+        let supportedAnswerFormats: [ORKAnswerFormat.Type] = [ORKTextChoiceAnswerFormat.self,
+                                                              ORKTextAnswerFormat.self,
+                                                              ORKBooleanAnswerFormat.self,
+                                                              ORKNumericAnswerFormat.self]
+        
+        let supportedQuestionTypes: [ORKQuestionType] = [.decimal,
+                                                         .integer,
+                                                         .text,
+                                                         .singleChoice,
+                                                         .multipleChoice,
+                                                         .boolean]
+        
+        if let formStep = step as? SBAFormStepProtocol,
+            let formItems = formStep.formItems {
+            
+            for item in formItems {
+                if let answerFormat = item.answerFormat {
+                    
+                    let formatOkay = supportedAnswerFormats.contains { (type) -> Bool in
+                        type == type(of: answerFormat)
+                    }
+                    
+                    let typeOkay = supportedQuestionTypes.contains(answerFormat.questionType)
+                    
+                    if !(formatOkay && typeOkay) {
+                        return false
+                    }
+                }
+            }
+        }
+        return true
+    }
+    
+    
+    // MARK: Initializers
+    
+    override public init(step: ORKStep, result: ORKResult?) {
+        super.init(step: step)
+        commonInit(with: result)        
+    }
+    
+    override public init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit(with: nil)
+    }
+    
+    func commonInit(with result: ORKResult?) {
+        setupModel(with: result)
+    }
+    
+    
+    // MARK: View lifecycle
+    
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // register for keyboard notifications
+        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardNotification(notification:)), name: NSNotification.Name.UIKeyboardWillChangeFrame, object: nil)
```

Comment:
Add in `viewWillAppear`. The design of the learn more may include a full screen modal in which case this will not be removed in `viewWillDisappear` but not added again after the learn more modal goes away.